lib/replace.ts

```diff
@@ -0,0 +1,104 @@
+import fs from "fs-extra";
+import { parse } from "@babel/parser";
+import traverse from "@babel/traverse";
+import * as t from "@babel/types";
+import { transformFromAst } from "@babel/core";
+
+async function replaceJSXTextInFile(
+  filePath: string,
+  replacement: { searchString: string; replaceWith: string }
+) {
+  const code = await fs.readFile(filePath, "utf-8");
+  const ast = parse(code, {
+    sourceType: "module",
+    plugins: ["jsx", "typescript"],
+  });
+
+  traverse(ast, {
+    JSXText(path) {
+      const { searchString, replaceWith } = replacement;
+
+      const regex = new RegExp(searchString, "gi");
+      if (regex.test(path.node.value)) {
+        const splitValues = splitByCaseInsensitive(
+          path.node.value,
+          searchString
+        );
+        const nodes: (t.JSXElement | t.JSXText)[] = [];
+
+        splitValues.forEach((splitValue) => {
+          if (splitValue.toLowerCase() === searchString.toLowerCase()) {
+            const identifier = t.jsxIdentifier("DittoComponent");
+            const componentId = t.jsxAttribute(
+              t.jsxIdentifier("componentId"),
+              t.stringLiteral(replaceWith)
+            );
+            const o = t.jsxOpeningElement(identifier, [componentId], true);
+            const jsxElement = t.jsxElement(o, undefined, [], true);
+            nodes.push(jsxElement);
+          } else {
+            nodes.push(t.jsxText(splitValue));
+          }
+        });
+
+        path.replaceWithMultiple(nodes);
+      }
+    },
+  });
+
+  // transfromFromAst types are wrong?
+  /* @ts-ignore */
+  const { code: transformedCode } = transformFromAst(ast, code, {
+    // Don't let this codebase's Babel config affect the code we're transforming.
+    configFile: false,
+  });
+  fs.writeFile(filePath, transformedCode);
+}
+
+function splitByCaseInsensitive(str: string, delimiter: string) {
+  return str.split(new RegExp(`(${delimiter})`, "gi")).filter((s) => s !== "");
+}
+
+function replace(options: string[]) {
+  let filePath: string;
+  let searchString: string;
+  let replaceWith: string;
+
+  try {
+    const parsedOptions = parseOptions(options);
+    filePath = parsedOptions.filePath;
+    searchString = parsedOptions.searchString;
+    replaceWith = parsedOptions.replaceWith;
+  } catch (e) {
+    console.error(e);
+    console.error(
+      "Usage for replace: ditto-cli replace <file path> <search string> <replace with>"
+    );
+    return;
+  }
+
+  replaceJSXTextInFile(filePath, { searchString, replaceWith });
+}
+
+function parseOptions(options: string[]): {
+  filePath: string;
+  searchString: string;
+  replaceWith: string;
+} {
+  if (options.length !== 3) {
+    throw new Error("The options array must contain exactly three strings.");
```

Comment:
Suggestion: could be helpful to ourselves to specify what the three strings should be here

Reply:
Good 👍